### ISSUE TITLE
`Style/FetchEnvVar` check for `if` condition

### ISSUE THREAD

**Issue by ShockwaveNN** [rubocop/rubocop#10569]
Not sure that `rubocop` can check it or dynamic check is required for such checks, but I'll report it anyway

## Steps to reproduce the problem

1. Create `test.rb` with content:
```ruby
# frozen_string_literal: true

if ENV['foo']
  puts 'some code'
  puts ENV['foo']
  puts 'other code'
end
```

2. run `rubocop -A` check
result code will be:
```ruby
# frozen_string_literal: true
    
if ENV['foo']
  puts 'some code'
  puts ENV.fetch('foo', nil)
  puts 'other code'
end
```

So fetch is literally useless in code like this

## RuboCop version

```
$ [bundle exec] rubocop -V
1.28.1 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 3.1.2 x86_64-linux)
  - rubocop-performance 1.13.3
```

**Comment by ShockwaveNN** [comment]
But in same time there may be code like this:
```ruby
if ENV['foo']
  ENV.delete('foo)
  puts 'some code'
  puts ENV['foo']
  puts 'other code'
end
```

And fetch is required in this code

**Comment by rthornton128** [comment]
`ENV['foo']`, `ENV.fetch('foo', nil)` and `ENV.dig('foo')` are all functionally equivalent. All three return `nil` when `ENV` does not contain the matching key. Perhaps what this cop should state, when recommending `ENV.fetch('foo', nil)` as an alternative, is the explicitness of returning `nil`?

Ultimately, a developer/team/project is free to ignore or disable cops. We should not blindly use all of them. If we think the cop has no value or does not apply to our code base, we should disable them. I think that applies to this case.
